sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLListener.scala

```diff
@@ -56,7 +56,10 @@ case class SparkListenerSQLExecutionStart(
 }
 
 @DeveloperApi
-case class SparkListenerSQLExecutionEnd(executionId: Long, time: Long)
+case class SparkListenerSQLExecutionEnd(
+    executionId: Long,
+    time: Long,
+    errorMessage: Option[String] = None)
```

Comment:
Thinking about this more, I think we should use `Some("")` to indicate no error, as None may mean event logs produced by old version of Spark that don't have this error message field. @ulysses-you